### ISSUE TITLE
Open global create when a user logs in for the first time

### ISSUE THREAD

**Issue by kevinksullivan** [Expensify/App#5047]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

# Problem

when a user first logs in to NewDot, there is no clear guidance on how to set up a Workspace or enable the Expensify Card, it's just a chat from Concierge (disregard the duplicate messages, separate issue being solved).

![image](https://user-images.githubusercontent.com/22508290/131935217-1c9b354e-74bf-4e31-9f0b-4d6182862a54.png)

### Actions Taken
1. Create a new account
2. Use the validation email to validate the account and set a password
3. Login

# Solution

When a new user signs into NewDot for the first time, let's open the Global Create menu so that it's clearer how to set up the Expensify Card or take any of the actions we are promoting. Let's show global create on the first sign in for both brand new Expensify users, and users who have an existing expensify account but haven't logged in to NewDot before.

![image](https://user-images.githubusercontent.com/22508290/131935582-29a31831-2755-4e46-89e1-b9df3cfbddbd.png)

![image](https://user-images.githubusercontent.com/22508290/131935610-0cf62a58-40ca-4c25-98cc-832076eb1b62.png)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @sophiepintoraetz (`AutoAssignerTriage`), see https://stackoverflow.com/c/expensify/questions/4749 for more details.

**Comment by sophiepintoraetz** [comment]
Lol, yes, I had thought this [ a lot of empty real estate] too when I did my one test of my N6 testing flow. Would we also use Concierge to prompt as well? Something like:
`"Click on the [whatever we're calling Global Create externally] to get started"`

**Comment by MitchExpensify** [comment]
Stepping in as its your weekend @sophiepintoraetz !

**Comment by MelvinBot** [comment]
Triggered auto assignment to @SofiedeVreese (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by MitchExpensify** [comment]
Going to grab for you and create the Upwork job @SofiedeVreese

**Comment by sophiepintoraetz** [comment]
Thanks Mitch! Was holding off because I'd asked a question [here](https://github.com/Expensify/App/issues/5047#issuecomment-912193091) but 👍

**Comment by MitchExpensify** [comment]
[Upwork job](https://www.upwork.com/jobs/~012f6bf2ece59e31fc)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @timszot (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by timszot** [comment]
I'm still ooo. Not sure why this was assigned to me

**Comment by Santhosh-Sellavel** [comment]
## Proposal
When the user signs up for the first time in here, `accountExists` is updated in ONYX with `ACCOUNT` key
https://github.com/Expensify/App/blob/b3421ceba196e5e2dd57b3fcbd4b30721650558c/src/libs/actions/Session.js#L123-L125

We could map the state from `ONYXKEYS.ACCOUNT`  to props of the  `SidebarScreen.js` component using `withONYX`
The in this block,
https://github.com/Expensify/App/blob/b3421ceba196e5e2dd57b3fcbd4b30721650558c/src/pages/home/sidebar/SidebarScreen.js#L52-L55

Include the below lines
```js

if (!this.props.account.accountExists) {
            this.toggleCreateMenu();

            // Reset the state since the create menu is already shown once.
            Onyx.merge(ONYXKEYS.ACCOUNT, {accountExists: true});
}

```
Note: We can create a new ONYX State if needed this state alone (I think its not necessary) , thanks!

**Comment by MelvinBot** [comment]
Triggered auto assignment to @Beamanator (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by Beamanator** [comment]
Hmm @Santhosh-Sellavel your idea sounds good except for one thing: The solution in the issue description says: 

> Let's show global create on the first sign in for both brand new Expensify users, **and users who have an existing expensify account but haven't logged in to NewDot before.**

The bolded part is missing from your proposal, I believe. If a user already created their account on E.com, `accountExists` will be `true` when logging in
<img width="760" alt="Screen Shot 2021-09-07 at 1 12 54 PM" src="https://user-images.githubusercontent.com/3885503/132340059-f85775da-083a-411e-bfca-c9670746fefc.png">

**Comment by Santhosh-Sellavel** [comment]
@Beamanator 
I get your point, I think I might need a little help here from the backend about the first new Expensify session. 

For most users,
I see the following options to do get a hook if there are no reports when we create a chat. Or when we have a concierge alone(It may show on every login). 

I think it's better to get help from the backend to ensure it happens only once.

**Comment by kevinksullivan** [comment]
Sorry for the late notice here @Santhosh-Sellavel . We concluded internally that we aren't set on this as a final solution, so we're going to close this for now.

**Comment by kevinksullivan** [comment]
And... this is coming back into play based on our [new approach](https://expensify.slack.com/archives/C020EPP9B9A/p1631132376089600) to global create. We're looking for the exact same solution as originally outlined in the OP. Thanks!

**Comment by mananjadhav** [comment]
@kevinksullivan  I don't think the link is accessible by contributors.

**Comment by Beamanator** [comment]
@mananjadhav you're right, the link is to an internal conversation, but the end result was basically as Kevin said - we're wanting to go with the original solution outlined in the OP :)

**Comment by Santhosh-Sellavel** [comment]
@Beamanator Quick question Is this possible without backend help?

**Comment by Beamanator** [comment]
Removing `Help Wanted` for now, will add it back once we're back with the new plan :)

**Comment by Beamanator** [comment]
OK so this ended up not needing any backend work! We should be able to accomplish what we need with these tools:

1. Let's create a new NVP called `IS_FIRST_TIME_NEW_EXPENSIFY_USER` (or if you have a better name, please suggest)
2. Use `NameValuePair.get` to check if `IS_FIRST_TIME_NEW_EXPENSIFY_USER` is `true` or not set
    - Most likely this will always be not set, so not sure if we ever need to check if this is `true`
3. If the above passes, automatically open global create.
4. Use `NameValuePair.set` to set `IS_FIRST_TIME_NEW_EXPENSIFY_USER` to `false` (only if it currently isn't `false`)

Now reviewing proposal for exactly how the implementation will look 👍

**Comment by mateomorris** [comment]
## Proposal
I'd add all that to `SignInPage.js`'s `componentDidMount`, and set an onyx store at the same time 
```
        const isNewAccount = NameValuePair.get(CONST.NVP.IS_FIRST_TIME_NEW_EXPENSIFY_USER)
        console.log({isNewAccount})

        if (isNewAccount === undefined || isNewAccount === true) {
            setFirstTimeUserNVP()
        } 
        Onyx.set(ONYXKEYS.IS_NEW_USER, isNewAccount);
```
That we can attach in `SidebarScreen.js` 
```
    withOnyx({
        betas: {
            key: ONYXKEYS.BETAS,
        },
        account: {
            key: ONYXKEYS.ACCOUNT
        }
```
and use in the constructor to set `isCreateMenuActive` 
```
        this.state = {
            isCreateMenuActive: !props.accountExists
        };
```

**Comment by Beamanator** [comment]
@mateomorris I like when your proposal is going! Although we'll need a few adjustments b/c `NameValuePair.get` doesn't return anything, and I believe we have some statement in our Philosophy explaining that actions should store data from the API into onyx, and shouldn't return anything. If I find this line I'll link it here :D

**Comment by mateomorris** [comment]
@Beamanator got it thanks, fixed!

**Comment by MitchExpensify** [comment]
Hired in Upwork @mateomorris !

**Comment by MitchExpensify** [comment]
when do you expect to have a PR up for this change @mateomorris ? Thanks for the proposal!

**Comment by mateomorris** [comment]
@MitchExpensify I take fridays off, so would tomorrow be okay?

**Comment by Beamanator** [comment]
I can probably review tomorrow or Sunday, will you be on Sat / Sun if I have comments over the weekend @mateomorris ?

**Comment by mateomorris** [comment]
@Beamanator for sure!

**Comment by Beamanator** [comment]
Great, if we could get it merged by Monday I think that would be fantastic 👍

**Comment by botify** [comment]
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

**Comment by botify** [comment]
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1230722409) to staging by @Beamanator in version: 1.0.97-1 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

**Comment by Santhosh-Sellavel** [comment]
@Beamanator should we show the global create on first login on new Expensify users only right.

Uninstalled the app/ install the app in new device/platform global create is triggered on login for even for existing new dot users.

**Comment by kevinksullivan** [comment]
>should we show the global create on first login on new Newdot users or expensify users never logged into newdot only right?

@Santhosh-Sellavel we should show this for expensify users logging into NewDot for the first time. So if they have logged in to OldDot before, but not NewDot, we show global create. However, if they have logged in on NewDot on one platform (i.e. mobile app), we wouldn't show it when they log in for the first time on a different platform (i.e. first session on web).

**Comment by kevinksullivan** [comment]
I also found that, when logging in for the first from OldDot by way of creating a workspace, I landed on the Workspace menu with global create open simultaneously.

![image](https://user-images.githubusercontent.com/22508290/133180677-08456ea9-4437-4ddc-adfd-66e1f016293b.png)


Are we able to update the order so we recognize that a workspace actually exists in these scenarios?

**Comment by Beamanator** [comment]
@Santhosh-Sellavel interesting find, I'll see if I can reproduce that!

@kevinksullivan good find a well, should the global create _not open automatically_ if the user has a workspace? A.k.a. if we fix showing the global create on that workspace screen (in your screenshot), then close the page, we shouldn't show the global create right?

**Comment by Beamanator** [comment]
@Santhosh-Sellavel do you have specific steps to reproduce what you're seeing? I uninstalled / installed on Android & signed in with my main New Expensify account and didn't see the global create show up, as expected 🤷 

@kevinksullivan I'm sure we can do this :D I will see if I can find a quick fix for that

**Comment by Santhosh-Sellavel** [comment]
@Beamanator it showed me only once, I see from the PR it's the expected outcome for all users to show once after the update.

**Comment by Beamanator** [comment]
Yep, cool thanks for confirming it's working as expected 🙏

**Comment by MitchExpensify** [comment]
@Beamanator 
> should the global create not open automatically if the user has a workspace? 

Yeah I don't think we should open global create if we're dropping new users into workspace configuration (i.e. when they have a workspace)

**Comment by Beamanator** [comment]
@MitchExpensify FYI the logic I implemented in my PR was to search for an existing workspace - if the user has a workspace (free policy), don't trigger opening global create, and still update the NVP @mateomorris  added so the global create won't be shown on future opens.

Another result of this change is: If someone currently uses New Expensify and hasn't loaded the app since this whole global create change was created, if they already have a workspace (no matter the flow they went through to create it), the global create **will not** show on login. I think this is still fine, since the purpose of auto-showing global create is to get more visibility and if they have already created it, they're aware of how to do it

**Comment by parasharrajat** [comment]
Is this expected? @Beamanator 
![image](https://user-images.githubusercontent.com/24370807/133472158-d16d2166-ee0a-4f5c-b0a9-7003363637ed.png)

**Comment by botify** [comment]
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

**Comment by botify** [comment]
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1239072912) to production by @francoisl in version: 1.0.98-1 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

**Comment by botify** [comment]
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1239082061) to staging by @robertjchen in version: 1.0.98-2 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

**Comment by botify** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.0.98-1 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pulls/5214\n

If no regressions arise, payment will be issued on 2021-09-27. :confetti_ball:

**Comment by mountiny** [comment]
This message is a bug, caused by today's deploy and some weird GH API behaviour. Updated the title back to what it was 🙇

**Comment by MitchExpensify** [comment]
payment made in Upwork
